### PR TITLE
*: collect key ranges for the read request in ResourceMeteringTag

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/11988.

What's Changed:

```commit-message
*: collect key ranges for the read request in ResourceMeteringTag
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

File: components/resource_metering/src/lib.rs [hunk 1]
```diff
@@ -280,6 +295,22 @@ impl TagInfos {
             store_id: peer.get_store_id(),
             peer_id: peer.get_id(),
             region_id: context.get_region_id(),
+            key_ranges: vec![],
+            extra_attachment: Vec::from(context.get_resource_group_tag()),
+        }
+    }
+
+    // create a TagInfos with start and end keys for an non-raw read request.
```
Comment (zhongzc) [inline components/resource_metering/src/lib.rs:303]:
```suggestion
    // create a TagInfos with start and end keys for a read request.
```

**Review by zhongzc — APPROVED** [review]

File: src/storage/tikv.code-workspace [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (disksing) [inline src/storage/tikv.code-workspace:1]: what's this file used for?

Comment (JmPotato) [reply src/storage/tikv.code-workspace]: Removed. PTAL.

**Review by disksing — APPROVED** [review]

**Pull request merged**